builtin/docker/builder.go

```diff
@@ -341,13 +344,30 @@ func (b *Builder) buildWithDocker(
 		return err
 	}
 
-	resp, err := cli.ImageBuild(ctx, buildCtx, types.ImageBuildOptions{
+	buildOpts := types.ImageBuildOptions{
 		Version:    ver,
 		Dockerfile: relDockerfile,
 		Tags:       []string{tag},
 		Remove:     true,
 		BuildArgs:  buildArgs,
-	})
+	}
+
+	// Buildkit builds need a session under most circumstances, but sessions are only supported in >1.39
+	if ver == types.BuilderBuildKit && versions.GreaterThanOrEqualTo(cli.ClientVersion(), "1.39") {
+
+		s, _ := session.NewSession(ctx, "waypoint", "")
+
+		dialSession := func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error) {
+			return cli.DialHijack(ctx, "/session", proto, meta)
+		}
+
+		go s.Run(ctx, dialSession)
+		defer s.Close()
+
+		buildOpts.SessionID = s.ID()
+	}
```

Comment:
We should probably log something here in the case where the CLI version isn't supported with sessions but buildkit was requested. It could be inside the StepGroup as a warning so it's visible, or maybe we could pass the `log` through to this func and just log a warning instead.